images/kubernetes-csi-external-provisioner/tests/02-deploy.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+# monopod:tag:k8s
+
+set -o errexit -o nounset -o errtrace -o pipefail -x
+
+function preflight() {
+  if [[ "${IMAGE_REGISTRY}" == "" ]]; then
+    echo "Must set IMAGE_REGISTRY environment variable. Exiting."
+    exit 1
+  fi
+
+  if [[ "${IMAGE_REPOSITORY}" == "" ]]; then
+    echo "Must set IMAGE_REPOSITORY environment variable. Exiting."
+    exit 1
+  fi
+
+  if [[ "${IMAGE_TAG}" == "" ]]; then
+    echo "Must set IMAGE_TAG environment variable. Exiting."
+    exit 1
+  fi
+}
+
+preflight
+
+function cleanup() {
+    # Print debug logs and status
+    kubectl get pods
+    kubectl describe pods
+    
+    # Seeing intermittent failures if we don't wait for a bit here
+    # The `rollout status`` below should wait for terminated pods to be removed
+    # However, we still occasionally see a terminating pod which fails when checking logs
+    sleep 10 
```

Comment:
This is annoying but it's been the only way I've seen the test consistently work.  The `kubectl rollout status` in the script should wait for old pods to be terminated but there's occasionally one that sticks around for a little while.  Other suggestions welcome else another sleep is the best I have ATM